sky/provision/kubernetes/instance.py

```diff
@@ -253,6 +253,31 @@ def _run_command_on_pods(node_name, node_namespace, command):
         _request_timeout=kubernetes.API_TIMEOUT)
     return cmd_output
 
+def _run_command_on_pods_with_logs(node_name, node_namespace, command, command_list):
+    cmd_output = kubernetes.stream()(
+        kubernetes.core_api().connect_get_namespaced_pod_exec,
+        node_name,
+        node_namespace,
+        command=command,
+        stderr=True,
+        stdin=True,
+        stdout=True,
+        tty=False,
+        _preload_content=False,
+        _request_timeout=kubernetes.API_TIMEOUT)
+    while cmd_output.is_open():
+        cmd_output.update(timeout=1)
+        if cmd_output.peek_stderr():
+            logger.warning(f"{cmd_output.read_stderr()}")
+        if command_list:
+            c = command_list.pop(0)
+            logger.debug(f"Running: {c}\n")
```

Comment:
Can we add an arg `operation_name: Optional[str] = None` and use that here to print nicer messages?  Also remove the `\n`
```suggestion
            log_text = operation_name + ': ' if operation_name else ''
            log_text += f"Running: {c}"
            logger.info(log_text)
```